### ISSUE TITLE
Permission error when creating files in mounted volume

### ISSUE THREAD

**Issue by qmeeus** [containers/libpod#3881]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**
I have a container which mounts a volume at runtime (using -v). I create a database and migration files in the container as user with uid > 10000. My host uid is < 10000. Therefore, I am having permission errors when those files are created. I am aware that setting my directory permissions to 777 would solve the problem but I'd rather not do that. I also know that I can use `--privileged` when I run a command in the container but this will just delay the problem...

**Steps to reproduce the issue:**
1. The command to run the container:
```
podman run -d \
    --name $container_name \
    --user $(id -u) \
    -v $(pwd)/api:/api \
    -v $(pwd)/data:/var/lib/sqlite \
    $custom_container
```
2. Then I execute some commands in the container, namely:
```
podman exec --user $(id -u) $container_name flask db init
podman exec --user $(id -u) $container_name flask db migrate
```
**Describe the results you received:**
Fails when trying to write in the current directory
`PermissionError: [Errno 13] Permission denied: 'migrations'`

**Describe the results you expected:**
If specifying my uid, I should not have permission errors

**Additional information you deem important (e.g. issue happens only occasionally):**
My container was build with my uid and runs as my uid, it works thus fine with docker.
**Output of `podman version`:**
```
Version:            1.4.4
RemoteAPI Version:  1
Go Version:         go1.12.6
OS/Arch:            linux/amd64
```

**Output of `podman info --debug`:**

```
debug:
  compiler: gc
  git commit: ""
  go version: go1.12.6
  podman version: 1.4.4
host:
  BuildahVersion: 1.9.0
  Conmon:
    package: Unknown
    path: /usr/bin/conmon
    version: 'conmon version 0.3.0, commit: 8455ce1ef385120deb827d0f0588c04357bad4c4'
  Distribution:
    distribution: manjaro
    version: unknown
  MemFree: 816041984
  MemTotal: 8264163328
  OCIRuntime:
    package: Unknown
    path: /usr/bin/runc
    version: |-
      runc version 1.0.0-rc8
      commit: 425e105d5a03fabd737a126ad93d62a9eeede87f
      spec: 1.0.1-dev
  SwapFree: 9185611776
  SwapTotal: 9446825984
  arch: amd64
  cpus: 4
  hostname: alice
  kernel: 5.2.8-1-MANJARO
  os: linux
  rootless: true
  uptime: 21h 32m 23.95s (Approximately 0.88 days)
registries:
  blocked: null
  insecure: null
  search:
  - docker.io
  - registry.fedoraproject.org
  - quay.io
  - registry.access.redhat.com
  - registry.centos.org
store:
  ConfigFile: /home/quent/.config/containers/storage.conf
  ContainerStore:
    number: 3
  GraphDriverName: vfs
  GraphOptions: null
  GraphRoot: /home/quent/.local/share/containers/storage
  GraphStatus: {}
  ImageStore:
    number: 4
  RunRoot: /run/user/1000
  VolumePath: /home/quent/.local/share/containers/storage/volumes
```

**Package info: output of `pacman -Si podman`:**

```
Repository      : community
Name            : podman
Version         : 1.4.4-1
Description     : Tool and library for running OCI-based containers in pods
Architecture    : x86_64
URL             : https://github.com/containers/libpod
Licenses        : Apache
Groups          : None
Provides        : None
Depends On      : cni-plugins  conmon  device-mapper  iptables  libseccomp  ostree  runc  skopeo  btrfs-progs  slirp4netns
Optional Deps   : podman-docker: for Docker-compatible CLI
Conflicts With  : None
Replaces        : None
Download Size   : 19,55 MiB
Installed Size  : 102,82 MiB
Packager        : Morten Linderud <foxboron@archlinux.org>
Build Date      : Mon 08 Jul 2019 11:48:39 CEST
Validated By    : MD5 Sum  SHA-256 Sum  Signature
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
```
$ uname -a
Linux 5.2.8-1-MANJARO #1 SMP PREEMPT Fri Aug 9 19:34:20 UTC 2019 x86_64 GNU/Linux
```

**Comment by mheon** [comment]
I assume you're running Podman as non-root?

If so, you might want to try removing the `--user` part of your command line. There are a few security benefits to running applications in rootless containers as UIDs other than 0, but they're not really that significant. UID 0 in the rootless container is the user that launched Podman, not real root on the system - it's not a privileged user, and has no permissions on the host that the user launching Podman did not have.

Alternatively, you can add `--userns=keep-id` and keep `--user`. This will map the UID running Podman so that the UID in the container matches the UID on the host.

**Comment by qmeeus** [comment]
> I assume you're running Podman as non-root?

Yes of course, all point of using podman ;-)

I rebuilt the container without creating a user, meaning that the container will run as root from now on. I also removed all mentions of `--user` in the run/exec commands. This seemed to solve the problem. The key was indeed NOT to create a user in my Dockerfile so that the container runs as root. Since I like to have the option to use both docker and podman, I will try the `--userns` option as well.

Thank you very much for your answer !
